src/Symfony/Component/Config/Definition/Builder/ArrayNodeDefinition.php

```diff
@@ -79,6 +84,20 @@ public function setBuilder(NodeBuilder $builder): void
         $this->nodeBuilder = $builder;
     }
 
+    /**
+     * Turns strings into arrays.
+     *
+     * @param list<ExprBuilder::TYPE_*> $allowedTypes
+     *
+     * @return $this
+     */
+    public function castToArray(array $allowedTypes, ?string $key = null): static
+    {
+        $this->allowedTypes = $allowedTypes;
+
+        return $this->beforeNormalization()->castToArray($key)->end();
```

Comment:
this non-conditional cast to array is _not_ a replacement for the existing normalization in most places where you started using it.

Reply:
OK, looking at the implementation of `castToArray`, it actually is a replacement, as it overrides any existing `ifPart` (meaning we have dead code).